client-src/default/index.js

```diff
@@ -8,28 +8,32 @@ const stripAnsi = require('strip-ansi');
 const log = require('loglevel').getLogger('webpack-dev-server');
 const socket = require('./socket');
 const overlay = require('./overlay');
-
-function getCurrentScriptSource() {
-  // `document.currentScript` is the most accurate way to find the current script,
-  // but is not supported in all browsers.
-  if (document.currentScript) {
-    return document.currentScript.getAttribute('src');
-  }
-  // Fall back to getting all scripts in the document.
-  const scriptElements = document.scripts || [];
-  const currentScript = scriptElements[scriptElements.length - 1];
-  if (currentScript) {
-    return currentScript.getAttribute('src');
-  }
-  // Fail as there was no script to use.
-  throw new Error('[WDS] Failed to get current script source.');
-}
+const sendMessage = require('./utils/sendMessage');
+const reloadApp = require('./utils/reloadApp');
+const getCurrentScriptSource = require('./utils/getCurrentScriptSource');
 
 let urlParts;
-let hotReload = true;
+const status = {
+  isUnloading: false,
+  currentHash: '',
+};
+const options = {
+  hot: false,
+  hotReload: true,
+  liveReload: false,
+  initial: true,
+  useWarningOverlay: false,
+  useErrorOverlay: false,
+  useProgress: false,
+};
+
+self.addEventListener('beforeunload', () => {
+  status.isUnloading = true;
+});
+
 if (typeof window !== 'undefined') {
   const qs = window.location.search.toLowerCase();
-  hotReload = qs.indexOf('hotreload=false') === -1;
+  options.hotReload = !qs.includes('hotreload=false');
```

Comment:
Can't be used, `includes` is not supported old browsers, we can change this only in major release

Reply:
You we need setup polyfill for this

Reply:
ok, I'll add babel-polyfill

Reply:
@hiroppy let's do this in other PR, using `babel-polyfill` directly can break many apps, because only one `babel-polyfill` can be defined, we should using `core-js` directly

Reply:
@hiroppy let's leave this as is in this PR and change this in other PR (where we include polyfills)